core/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java

```diff
@@ -214,6 +215,32 @@ public void testChangeFormat() throws IOException {
         assertEquals(1457654400000L, pointField.numericValue().longValue());
     }
 
+    public void testFloatEpochFormat() throws IOException {
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type")
+                .startObject("properties").startObject("field").field("type", "date")
+                .field("format", "epoch_millis").endObject().endObject()
+                .endObject().endObject().string();
+
+        DocumentMapper mapper = parser.parse("type", new CompressedXContent(mapping));
+
+        assertEquals(mapping, mapper.mappingSource().toString());
+
+        long millisFromEpoch = randomNonNegativeLong();
+        String epochFloatValue = String.format(Locale.US, "%d.%d", millisFromEpoch, randomNonNegativeLong());
```

Comment:
Maybe also randomly append a negative prefix to also test parsing negative values here?